sources/packages/backend/apps/api/src/route-controllers/student-loan-balance/student-loan-balance.aest.controller.ts

```diff
@@ -40,14 +46,17 @@ export class StudentLoanBalanceAESTController extends BaseController {
     if (!studentExist) {
       throw new NotFoundException("Student not found.");
     }
-    //TODO: Remove the static values when actual implementation is done.
-    const staticData: StudentLoanBalanceAPIOutDTO = {
-      loanBalanceDetails: [
-        { balanceDate: "2024-03-10", cslBalance: 100 },
-        { balanceDate: "2024-02-10", cslBalance: 200 },
-        { balanceDate: "2024-01-10", cslBalance: 300 },
-      ],
+    const studentLoanBalances =
+      await this.studentLoanBalanceService.getStudentLoanBalance(studentId);
+    const loanBalanceDetails =
+      studentLoanBalances.map<StudentLoanBalanceDetailAPIOutDTO>(
```

Comment:
Is there a benefit to declaring the type explicitly here?

Reply:
As it is not the direct return object, (map creates an array which is returned inside a property `loanBalanceDetails`) the intellisense is not showing the properties without explicitly using the type.

![image](https://github.com/bcgov/SIMS/assets/54600590/eedc83b8-c573-4da8-862b-f7f9741266a0)